h2/src/main/org/h2/mode/Regclass.java

```diff
@@ -39,6 +39,10 @@ public Value getValue(Session session) {
         if (value == ValueNull.INSTANCE) {
             return ValueNull.INSTANCE;
         }
+        int valueType = value.getValueType();
+        if (valueType >= Value.TINYINT && valueType <= Value.BIGINT) {
+            return value;
+        }
```

Comment:
This expression has declared type `INTEGER` and may return only `ValueNull.INSTANCE` and instances of `ValueInteger`.

You can use `convertToInt(null)`, but most likely the 32-bit unsigned values need special handling here to avoid `NUMERIC_VALUE_OUT_OF_RANGE_2` error.

Reply:
Can we just truncate to lower-32-bit INTEGER for BIGINT?
```
        if (valueType >= Value.TINYINT && valueType <= Value.INTEGER) {
            return value.convertToInt(null);
        }
        if (valueType == Value.BIGINT) {
            return ValueInteger.get((int) value.getLong());
        }
```

Reply:
I think we can.